tests/reporter/kubernetes.go

```diff
@@ -0,0 +1,221 @@
+package reporter
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/types"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+)
+
+type KubernetesReporter struct {
+	failureCount int
+	artifactsDir string
+	mux          sync.Mutex
+}
+
+func NewKubernetesReporter(artifactsDir string) *KubernetesReporter {
+	return &KubernetesReporter{
+		failureCount: 0,
+		artifactsDir: artifactsDir,
+	}
+}
+
+func (r *KubernetesReporter) SpecSuiteWillBegin(config config.GinkgoConfigType, summary *types.SuiteSummary) {
+
+}
+
+func (r *KubernetesReporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {
+	// clean up artifacts from previous run
+	if r.artifactsDir != "" {
+		os.RemoveAll(r.artifactsDir)
+	}
+}
+
+func (r *KubernetesReporter) SpecWillRun(specSummary *types.SpecSummary) {
+}
+
+func (r *KubernetesReporter) SpecDidComplete(specSummary *types.SpecSummary) {
+	r.mux.Lock()
```

Comment:
Why the need for the mutex? Are the specs expected to be executed in parallel?

Reply:
In theory. It was cheap to add. I can also remove it.

Reply:
and might be that ginkgo itself protects it too already.

Reply:
Yes, my main concern here is that if they run in parallel, the first report passes the mutex, the other ones locks on it and the collected info is a mixture of all the specs blocked on the mutex.

Reply:
So there is only one instance of the reporter and only this one log. Every invocation has a unique file-prefix, to have individual files for each test which fails.

Reply:
Yes, but the reporter collects all the info of all the vmis, pod, around so if we had multiple specs at the same time the reported logs would collect the infos of all the specs locked on the mutex, isn't it?

Reply:
*in the same time window

Reply:
ah, absolutely right. I think it is not that bad, because they both see a similar cluster state